metagraph/src/annotation/annotation_converters.cpp

```diff
@@ -751,6 +786,602 @@ convert_to_RbBRWT<RbBRWTAnnotator>(const std::vector<std::string> &annotation_fi
     return std::make_unique<RbBRWTAnnotator>(std::move(matrix), label_encoder);
 }
 
+void convert_batch_to_row_disk(
+        const std::function<bool(const std::vector<std::string> &, const ColumnCallback &, size_t)>
+                &get_cols,
+        const std::vector<std::string> &source_files,
+        const std::string &outfname,
+        const std::function<void(std::ofstream &)> &decorate,
+        uint64_t num_rows,
+        size_t num_threads) {
+    std::vector<std::string> col_names;
+    std::vector<std::unique_ptr<bit_vector>> columns;
+    uint64_t num_set_bits {};
+    std::mutex mtx;
+    bool success = get_cols(
+            source_files,
+            [&](uint64_t column_index, const std::string &label,
+                std::unique_ptr<bit_vector> &&column) {
+                std::lock_guard<std::mutex> lck(mtx);
+
+                if (num_rows != column->size()) {
+                    logger->error("Annotations have different number of rows");
+                    std::exit(1);
+                }
+
+                num_set_bits += column->num_set_bits();
+                if (columns.size() <= column_index) {
+                    columns.resize(column_index + 1);
+                    col_names.resize(column_index + 1);
+                }
```

Comment:
This would cause a race condition. I wonder why this is never caught in unit tests. Do we test parallel conversion?

Reply:
Ah, I see a lock at the top now. All good.